src/state/apis/snapshot/snapshot.ts

```diff
@@ -152,6 +194,63 @@ export const snapshotApi = createApi({
         }
       },
     }),
+    getThorVotingPower: build.query<FoxVotingPowerCryptoBalance, void>({
+      queryFn: async (_, { dispatch, getState }) => {
+        try {
+          const accountIds: AccountId[] =
+            (getState() as ReduxState).portfolio.accountMetadata.ids ?? []
+          const strategies = await (async () => {
+            const maybeSliceStragies = (getState() as ReduxState).snapshot.thorStrategies
+            if (maybeSliceStragies) return maybeSliceStragies
+
+            const strategiesResult = await dispatch(
+              snapshotApi.endpoints.getThorStrategies.initiate(),
+            )
+            return strategiesResult?.data
+          })()
+          if (!strategies) {
+            console.log('snapshotApi getThorVotingPower could not get strategies')
+            return { data: bn(0).toString() }
+          }
+          const evmAddresses = Array.from(
+            accountIds.reduce<Set<string>>((acc, accountId) => {
+              const { account, chainId } = fromAccountId(accountId)
+              isEvmChainId(chainId) && acc.add(account)
+              return acc
+            }, new Set()),
+          )
+          const delegation = false // don't let people delegate for discounts - ambiguous in spec
+          const votingPowerResults = await Promise.all(
```

Comment:
We absolutely want a partial success which should already be the case

This is also the case for FOX voting power, should we do both in a follow up or partially do it in this one and include fox in another one?

Reply:
If any of the `getVotingPower` calls throw the `Promise.all` will also throw, so if we want partial failure/success we can use `Promise.allSettled` combined with introspection if thats what we want

Reply:
Yep it would make sense, but I would have to also modify the fox power side which is unrelated to this PR, should I do it in this one or reserve it for a follow up?

Reply:
Changed both in [3ae7cc6](https://github.com/shapeshift/web/pull/8044/commits/3ae7cc687304721da8a65d8b8a8c60ba663fdada)